tests/contrib/psycopg/test_psycopg.py

```diff
@@ -192,71 +242,38 @@ def test_connect_factory(self):
         # ensure we have the service types
         service_meta = tracer.writer.pop_services()
         expected = {
-            "db" : {"app":"postgres", "app_type":"db"},
-            "another" : {"app":"postgres", "app_type":"db"},
+            'db' : {'app':'postgres', 'app_type':'db'},
+            'another' : {'app':'postgres', 'app_type':'db'},
         }
-        eq_(service_meta, expected)
+        self.assertEquals(service_meta, expected)
 
 
-class TestPsycopgPatch(PsycopgCore):
-
-    def setUp(self):
-        patch()
-
-    def tearDown(self):
-        unpatch()
-
-    def _get_conn_and_tracer(self):
-        conn = psycopg2.connect(**POSTGRES_CONFIG)
-        tracer = get_dummy_tracer()
-        Pin.get_from(conn).clone(tracer=tracer).onto(conn)
-
-        return conn, tracer
-
-    def test_patch_unpatch(self):
-        tracer = get_dummy_tracer()
-        writer = tracer.writer
-
-        # Test patch idempotence
-        patch()
-        patch()
-
-        service = "fo"
-
-        conn = psycopg2.connect(**POSTGRES_CONFIG)
-        Pin.get_from(conn).clone(service=service, tracer=tracer).onto(conn)
-        conn.cursor().execute("select 'blah'")
-
-        spans = writer.pop()
-        assert spans, spans
-        eq_(len(spans), 1)
-
-        # Test unpatch
-        unpatch()
-
-        conn = psycopg2.connect(**POSTGRES_CONFIG)
-        conn.cursor().execute("select 'blah'")
-
-        spans = writer.pop()
-        assert not spans, spans
-
-        # Test patch again
-        patch()
+    @skipIf(PSYCOPG_VERSION < (2, 7), 'SQL string composition not available in psycopg2<2.7')
+    def test_composed_query(self):
+        """ Checks whether execution of composed SQL string is traced """
+        query = SQL(' union all ' ).join(
+            [SQL("""select 'one' as x"""),
+             SQL("""select 'two' as x""")])
+        db, tracer = self._get_conn_and_tracer()
 
-        conn = psycopg2.connect(**POSTGRES_CONFIG)
-        Pin.get_from(conn).clone(service=service, tracer=tracer).onto(conn)
-        conn.cursor().execute("select 'blah'")
+        with db.cursor() as cur:
+            cur.execute(query=query)
+            rows = cur.fetchall()
+            assert len(rows) == 2, rows
+            assert rows[0][0] == 'one'
+            assert rows[1][0] == 'two'
 
-        spans = writer.pop()
-        assert spans, spans
-        eq_(len(spans), 1)
+        spans = tracer.writer.pop()
+        assert len(spans) == 1
+        span = spans[0]
+        self.assertEquals(span.name, 'postgres.query')
```

Comment:
we probably want an assertion on the resource here too right? It should be the same as the `query.as_string(...)` right?